etc/di.xml

```diff
@@ -254,4 +211,35 @@
             </argument>
         </arguments>
     </type>
+    <type name="Nosto\Tagging\Model\Service\Sync\Upsert\SyncService">
+        <arguments>
+            <argument name="productService" xsi:type="object" shared="false">
+                Nosto\Tagging\Model\Service\Product\DefaultProductService
```

Comment:
Maybe we should use the CachingProductService here?

Reply:
We cannot do that. With the current logic, if we'd use `CachingProductService` here the cache would not get updated ever.